### DISCUSSION TITLE
Failed to fetch device status for ... (even for permanently powered WiFi devices)

### DISCUSSION THREAD

**Discussion by TheS1ngularity** [make-all/tuya-local#2197] [Q&A]
Hi guys, 
I was really glad to see that local control can be achieved for Tuya devices. This is really a great project!
I have some Tuya smart devices, some permanently powered: smart plugs, dimmers, pet feeder. And some that are sometimes turned off: smart lights.

Initially I've set everything up in the cloud integration, but, ofc, the ideal scenario is to have no dependency on the cloud, so when I found out about this project, I've set everything up in the Tuya Local and disabled the cloud integration completely.
I have all the devices with static IP allocation and I am not using the SmartLife app (force stopped) and not in use at all. 

I was having spam of errors in HA: e.g. ~18000 errors in 10-12 hours from 18 devices). I think this was also causing my HA OS VM to crash (not sure why though). I realized that devices that are not permanently on might cause this (more than the others), so I removed those from Tuya Local and used them in the cloud integration. Also in the cloud integration I disabled the other devices (since I had all of them in initially) so they're not active on both cloud and Tuya Local integrations. 

My problem is still there though, most of my permanently powered devices are still spamming the error. I now have 8 devices in Tuya Local and 4 of them have the spam error (cleared the log before posting here):

```
Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:564
integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 12:55:41 (177 occurrences)
Last logged: 13:12:31
```

The other 4 don't seem to have the problem though. So I'm wondering why this would be the case? What's the behavior of these devices that leads to this error and why so often? Anyone has some insights into this and a potential fix?
In the meantime I'll probably keep just the 4 stable devices in Tuya Local and continue with everything else in the Tuya cloud integration, but I'd definitely love to have everything running locally.

Cheers in advance!

**Comment by make-all** [comment]
You didn't include the actual error message with your log snippet, but assuming it is the one in the title, this can be caused by mismatched protocol, or unstable WiFi.

**Comment by TheS1ngularity** [comment]
Hi, thanks for the reply. Wifi router is very close to the devices, they don't have issues when used via cloud.

How can I check the protocol supported by devices? I managed to do it once, but don't recall where I found the tutorial :( Would appreciate if you can point me in the right direction or maybe add it in the readme, even nicer.

**Reply by MolloPaul** [reply]
Use TinyTura see [https://github.com/jasonacox/tinytuya]
